src-tauri/src/compressor.rs

```diff
@@ -1,12 +1,182 @@
 }
 
 pub struct VoidCompressor {}
+
 impl Compressor for VoidCompressor {
-    fn compress(&self, input: Vec<u8>) -> Vec<u8> {
-        todo!()
+    fn compress(&self, inpath: PathBuf) -> Result<PathBuf> {
+        // Currently, this function just aliases to compress_lz4. May use patterm matching to select a function in future.
+        //
+        //  match inpath.extension().and_then(|os_str| os_str.to_str()) {
+        //      Some("txt") | Some("org") | Some("md") => self.compress_llm(inpath),
+        //      _ => self.compress_lz4(inpath),
+        //  }
+        dbg!("here");
+        self.compress_lz4(inpath)
+    }
+
+    // Use ts_zip and pipe output to mut progress
+    // Progress display code from https://stackoverflow.com/questions/31576555/unable-to-pipe-to-or-from-spawned-child-process-more-than-once/31577297#31577297, untested
+    fn compress_llm(&self, inpath: PathBuf) -> Result<PathBuf> {
+        let filename = inpath.file_stem().unwrap();
+        let outpath = Path::new("/path/to/data/dir")
+            .join(filename)
+            .with_extension("bin");
+
+        let mut process = Command::new("/path/to/ts_zip")
+            .arg("-m")
+            .arg("path/to/model")
+            .arg("c")
+            .arg(inpath.as_path().to_str().unwrap())
+            .arg(outpath.as_path().to_str().unwrap())
+            .stdout(Stdio::piped())
+            .spawn()?;
+
+        let stdout = process.stdout.take().unwrap();
+        let mut reader = BufReader::new(stdout);
+        let mut progress_display = String::new();
+        let mut buffer = String::new();
+
+        loop {
+            match reader.read_line(&mut buffer) {
+                Ok(0) => break,
+                Ok(_) => {
+                    if buffer.contains("ratio") {
+                        break;
+                    }
+                    progress_display = buffer.clone();
+                    println!("{}", progress_display);
+                    buffer.clear();
+                }
+                Err(e) => {
+                    return Err(std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        format!("Error reading from ts_zip: {}", e),
+                    ));
+                }
+            }
+        }
+
+        let output = process.wait()?;
+        if !output.success() {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "ts_zip compression failed",
+            ));
+        }
+
+        Ok(outpath)
+    }
+
+    fn compress_lz4(&self, inpath: PathBuf) -> Result<PathBuf> {
+        dbg!("here");
+        let mut input_file = File::open(inpath.as_path())?;
+        dbg!("here");
+        let filename = inpath.file_stem().unwrap();
+        dbg!("here");
+        let out_dir = "compressed_lz4/";
+        fs::create_dir_all(out_dir);
+        let outpath = Path::new(out_dir).join(filename).with_extension("lz4");
```

Comment:
The assumption that this directory exists causes an error to be thrown; additionally, there is no handling of the case where the directory does not exist; I pushed a way to fix this, but this assumption is made in several other places in the code; please fix them and add proper handling of the case when the directory doesn't exist (either make it through Rust, or prompt the user to make it themselves).